internal/host/validator.go

```diff
@@ -1829,3 +1831,62 @@ func (v *validator) inventoryHasIP(inventory *models.Inventory, ipAddress string
 	}
 	return false, nil
 }
+
+func (v *validator) iSCSIHostNetworkInterfaceDoesNotBelongToMachineCidr(c *validationContext) (ValidationStatus, string) {
+	installationDisk, err := hostutil.GetHostInstallationDisk(c.host)
+	if err != nil || installationDisk == nil {
+		return ValidationSuccessSuppressOutput, ""
+	}
+
+	if installationDisk.DriveType != models.DriveTypeISCSI {
+		// Validation is not relevant in this case
+		return ValidationSuccessSuppressOutput, ""
+	}
+
+	if c.inventory == nil || !network.IsMachineCidrAvailable(c.cluster) {
+		return ValidationPending, "Missing inventory or machine network CIDR"
+	}
+
+	if installationDisk.Iscsi == nil {
+		// If this is nil, the disk shouldn't have passed the eligilibily test in the first place
+		v.log.Warn("iSCSI installation disk is missing host IP address")
+		return ValidationError, "iSCSI installation disk is missing host IP address"
+	}
+
+	nic, err := network.FindInterfaceByIPString(installationDisk.Iscsi.HostIPAddress, c.inventory.Interfaces)
+	if err != nil {
+		v.log.WithError(err).Warn("Cannot find network interface associated to iSCSI host IP address")
+		return ValidationError, "Cannot find network interface associated to iSCSI host IP address"
+	}
+
+	// Look if one of the IP set on the interface connected to the iSCSI disk belong to machine network CIDRs
+	ips := append(nic.IPV4Addresses, nic.IPV6Addresses...)
+	_, found := lo.Find(ips, func(ip string) bool {
```

Comment:
Maybe we could extract this function to make it both clearer to the reader and more testeable (as commented in the test we could test edge cases in the function test, and more generic tests in the validation test)

Reply:
it got refactored with functions in network packages